telegram-bot/newUser.go

```diff
@@ -0,0 +1,130 @@
+		"what do you want to learn?",
+		"what is your current operating system?",
+		"if you are new to open source.",
+		"which technology fascinates you the most?",
+		"what have you been exploring recently?",
+	)
+
+	collection := client.Database("test").Collection("user")
+	data := newUser{
+		UserName:     user.UserName,
+		FirstName:    user.FirstName,
+		UserID:       user.ID,
+		Introduction: false,
+		JoinDate:     time.Now(),
+	}
+	log.Println(data)
+	check, err := collection.InsertOne(context.TODO(), data)
+	log.Println(check)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	rand.Seed(time.Now().UnixNano())
+	min := 0
+	max := len(botSlice) - 1
+	randomNum1 := (rand.Intn(max-min+1) + min)
+	randomNum2 := (rand.Intn(max-min+1) + min)
+	var welcomeMessage1 = fmt.Sprintf(botSlice[randomNum1])
+	var welcomeMessage2 = fmt.Sprintf(quesSlice[randomNum2])
+	reply := tbot.NewMessage(ID, "Welcome "+User+", I am the OSDC-bot, "+welcomeMessage1+
+		" Please introduce yourself. To start with, you can tell us "+welcomeMessage2+" Start your message with /introduction so that I can verify you.")
+	reply.ParseMode = "markdown"
+	bot.Send(reply)
+}
+
+func introverify(user *tbot.User, ID int64, client mongo.Client) {
+	collection := client.Database("test").Collection("user")
+	filter := bson.M{"userid": user.ID, "introduction": false}
+	update := bson.M{"$set": bson.M{"introduction": true}}
+	result, err := collection.UpdateOne(context.TODO(), filter, update)
+	if err != nil {
+		// ErrNoDocuments means that the filter did not match any documents in the collection
+		if err == mongo.ErrNoDocuments {
+			log.Println("no documents")
+			bot.Send(tbot.NewMessage(ID, "No need to introduce again, I already know you."))
+			return
+		}
+		log.Fatal(err)
+	}
+	if result.MatchedCount > 0 {
+		bot.Send(tbot.NewMessage(ID, "Thanks for introducing yourself, You are now a verified member of OSDC :) "))
+	} else {
+		bot.Send(tbot.NewMessage(ID, "No need to introduce again, I already know you."))
+	}
+
+}
+
+func introkick(ID int64, client mongo.Client) {
+	collection := client.Database("test").Collection("user")
+	ctx := context.Background()
+	cursor, err := collection.Find(context.TODO(), bson.D{})
+	if err != nil {
+		log.Fatal(err)
+		defer cursor.Close(ctx)
+	} else {
+		// iterate over docs using Next()
+		for cursor.Next(ctx) {
+			// Declare a result BSON object
+			var result newUser
+			err := cursor.Decode(&result)
+			if err != nil {
+				log.Fatal(err)
+			} else {
+				// log.Println(time.Now().Local().Day(), result.JoinDate.Day())
+				if result.Introduction == false {
+					curHour := time.Now().Local().Hour()
+					curDay := time.Now().Local().Day()
+					User := fmt.Sprintf("[%v](tg://user?id=%v)", result.FirstName, result.UserID)
+					log.Println(curDay-result.JoinDate.Day(), curHour-result.JoinDate.Hour())
+					if (curDay-result.JoinDate.Day() > 0 && curHour-result.JoinDate.Hour() > 0) || curDay-result.JoinDate.Day() > 1 {
```

Comment:
Does the `Day` return the date? In that case, if `curDay` is 1 and `JoinDate.Day()` is 31. The result may be negative. 

Try taking the mod of difference if that's the case.